### PR TITLE
Executable refactor

### PR DESCRIPTION
**Pull request by NicolaMirchev**
**Description**:
Massive sdk refactor of executable structure:
- This is the new version of code, where we have reduced a lot of the code duplication especially for `Execute` inside all transaction instances and quieries. 
- In this commit we do a `major change` regarding the return type from `TransactionFromBytes(data[] bytes) (interface{}, error)`. We had to make this function return a reference to an instance(*{some}Transaction), instead of a value{{some}Transaction}, because of the new design and the interface variable inside "Transaction".

Details here:
https://www.notion.so/limechain/Hedera-Go-SDK-Executable-Architecture-Refactor-f5f6bc945fe24b99a7d6e7f9cf3ff605

**Related issue(s)**:
#832 
#833
#834
#835

**Notes for reviewer**:

**Checklist**

- [x] Documented (Notion)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: account_allowance_adjust_transaction_unit_test.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func TestUnitAccountAllowanceAdjustTransactionGet(t *testing.T) {
 	require.Equal(t, 1, tx.GetMaxRetry())
 	require.Equal(t, time.Second*120, tx.GetMaxBackoff())
 	require.Equal(t, time.Second*1, tx.GetMinBackoff())
-	require.Equal(t, fmt.Sprintf("AccountAllowanceAdjustTransaction:%v", transactionID.ValidStart.UnixNano()), tx._GetLogID())
+	require.Equal(t, fmt.Sprint("AccountAllowanceAdjustTransaction"), tx.getName())
```
Comment (gsstoykov) [inline account_allowance_adjust_transaction_unit_test.go:100]: Consider not using `fmt.Sprint`.

File: transaction.go [hunk 2]
```diff
@@ -4705,11 +4672,178 @@ func TransactionExecute(transaction interface{}, client *Client) (TransactionRes
...
+}
+
+// Building empty object as "default" implementation. All inhertents must implement their own implementation.
+func (tx *Transaction) getName() string {
+	return "transaction"
+}
+
+func (tx *Transaction) getLogID(transactionInterface Executable) string {
+	timestamp := tx.transactionIDs._GetCurrent().(TransactionID).ValidStart
+	return fmt.Sprintf("%s:%d", transactionInterface.getName(), timestamp.UnixNano())
```
Comment (gsstoykov) [inline transaction.go:4744]: Consider not using `fmt.Sprintf`.

File: transaction.go [hunk 3]
```diff
@@ -561,65 +533,9 @@ func (this *Transaction) _KeyAlreadySigned(
...
-}
-
-// String returns a string representation of the Transaction
-func (this *Transaction) String() string {
-	switch sig := this.signedTransactions._Get(0).(type) { //nolint
+// String returns a string representation of the transaction
+func (tx *Transaction) String() string {
+	switch sig := tx.signedTransactions._Get(0).(type) { //nolint
 	case *services.SignedTransaction:
 		return fmt.Sprintf("%+v", sig)
```
Comment (gsstoykov) [inline transaction.go:540]: Consider not using `fmt.Sprintf`.

File: errors.go [hunk 4]
```diff
@@ -74,14 +74,6 @@ type ErrMaxQueryPaymentExceeded struct {
...
-// 	return ErrMaxQueryPaymentExceeded{
-// 		QueryCost:       queryCost,
-// 		MaxQueryPayment: maxQueryPayment,
-// 		query:           reflect.TypeOf(*transaction).Name(),
-// 	}
-// }
-
 // Error() implements the Error interface
 func (e ErrMaxQueryPaymentExceeded) Error() string {
 	return fmt.Sprintf("cost of %s (%s) without explicit payment is greater than the max query payment of %s",
```
Comment (gsstoykov) [inline errors.go:79]: Consider not using `fmt.Sprintf`.

File: custom_fixed_fee.go [hunk 5]
```diff
@@ -178,7 +178,7 @@ func (fee CustomFixedFee) ToBytes() []byte {
 }
 
 // String returns a string representation of the CustomFixedFee
-func (fee CustomFixedFee) String() string {
+func (fee *CustomFixedFee) String() string {
 	if fee.DenominationTokenID != nil {
 		return fmt.Sprintf("feeCollectorAccountID: %s, amount: %d, denominatingTokenID: %s", fee.FeeCollectorAccountID.String(), fee.Amount, fee.DenominationTokenID.String())
```
Comment (gsstoykov) [inline custom_fixed_fee.go:183]: Consider not using `fmt.Sprintf`.

File: account_allowance_approve_transaction.go [hunk 6]
```diff
@@ -21,15 +21,14 @@ package hedera
...
-	"fmt"
 	"time"
 
 	"github.com/hashgraph/hedera-protobufs-go/services"
 )
 
 // AccountAllowanceApproveTransaction
 // Creates one or more hbar/token approved allowances <b>relative to the owner account specified in the allowances of
-// this transaction</b>. Each allowance grants a spender the right to transfer a pre-determined amount of the owner's
+// tx transaction</b>. Each allowance grants a spender the right to transfer a pre-determined amount of the owner's
```
Comment (gsstoykov) [inline account_allowance_approve_transaction.go:31]: Bold tag redundant here.

**Review by rokn — APPROVED** [review]

**Pull request merged**